### PR TITLE
fix: #1238, fix 404 when base config is not '/'

### PR DESCRIPTION
**Pull request by BuptStEve**
**Summary**

when you set base config, you will get a `404` error

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

**Comment by BuptStEve** [comment]
https://github.com/vuejs/vue-cli/blob/9c1e797ac6c25b5827403693e018eb199300d067/packages/%40vue/cli-service/lib/commands/serve.js#L140-L145

File: packages/@vuepress/core/lib/webpack/createBaseConfig.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ module.exports = function createBaseConfig ({
     .output
       .path(outDir)
       .filename(isProd ? 'assets/js/[name].[chunkhash:8].js' : 'assets/js/[name].js')
-      .publicPath(isProd ? publicPath : '/')
+      .publicPath(publicPath)
```
Comment (ulivz) [inline packages/@vuepress/core/lib/webpack/createBaseConfig.js]: What's the reason of changing it?

Comment (BuptStEve) [reply packages/@vuepress/core/lib/webpack/createBaseConfig.js]: https://github.com/vuejs/vuepress/issues/1238

Comment (BuptStEve) [reply packages/@vuepress/core/lib/webpack/createBaseConfig.js]: you can reproduce it by this repo
https://github.com/BuptStEve/vuepress-bug

File: packages/@vuepress/core/lib/dev.js [hunk 2]
```diff
@@ -124,14 +126,19 @@ async function prepareServer (sourceDir, cliOptions = {}, context) {
...
+        { from: /./, to: path.posix.join(ctx.base, 'index.html') }
       ]
     },
     overlay: false,
     host,
-    contentBase: path.resolve(sourceDir, '.vuepress/public'),
+    contentBase,
     before (app, server) {
+      if (fs.existsSync(contentBase)) {
+        app.use(ctx.base, require('express').static(contentBase))
```
Comment (ulivz) [inline packages/@vuepress/core/lib/dev.js]: contentBase is implemented by `require('express').static`.

**Pull request merged**